### ISSUE TITLE
[go] mingw-w64-x86_64-go bad GOPATH default

### ISSUE THREAD

**Issue by nerdralph** [msys2/MINGW-packages#8437]
```
# go version
go: cannot find GOROOT directory: D:\a\_temp\msys\msys64\mingw64\lib\go
```

work around:
export GOROOT=/mingw64/lib/go

**Comment by lazka** [comment]
I can't reproduce. Do you have a "D:" drive by any chance? maybe that triggers it.

**Comment by naveen521kk** [comment]
See https://github.com/msys2/MINGW-packages/issues/7844 and https://github.com/msys2/MINGW-packages/issues/421

**Comment by nerdralph** [comment]
> 
> 
> I can't reproduce. Do you have a "D:" drive by any chance? maybe that triggers it.

Yes, though @naveen521kk comments indicate this has been a long-running issue.

**Comment by 1480c1** [comment]
I can still reproduce this when not loading any profiles (aka from not bash)
```bash
PS C:\Users\cddeg> C:\msys64\ucrt64\bin\go.exe install -v golang.org/x/tools/cmd/goimports@latest
go: cannot find GOROOT directory: D:\a\msys64\ucrt64\lib\go
```

**Comment by 1480c1** [comment]
Perhaps the three issues should be consolidated? It seems to stem primarily from the fact that the GOROOT's default value is hardcoded at buildtime rather than run-time, perhaps something similar to pkg-config's relocation could be done

**Comment by RivenSkaye** [comment]
I see someone else mentioned 421 and 7844 already, should two of them be closed as a duplicate similar to #665?
As I commented on 421 when I stumbled across it, the issue persists still even from a loaded profile

**Comment by thewh1teagle** [comment]
> I can't reproduce. Do you have a "D:" drive by any chance? maybe that triggers it.

I experience the same issue. I do have D: drive

**Comment by mlt** [comment]
Can we patch go to deduce absolute path at runtime from `$MINGW_PREFIX`?
